source/slides/01_linux_basics.rst

```diff
@@ -0,0 +1,506 @@
+    -rw-rw-r--      1 test    test         0    Nov 13 14:09 file.txt
+    drwxrwxr-x      2 test    test      4096    Nov  6 13:22 Pictures
+    ----------     -------  -------  -------- ------------ -------------
+        |             |        |         |         |             |
+        |             |        |         |         |         File Name
+        |             |        |         |         +---  Modification Time
+        |             |        |         +-------------   Size (in bytes)
+        |             |        +-----------------------        Group
+        |             +--------------------------------        Owner
+        +----------------------------------------------   File Permissions
+
+``-`` is a normal file
+
+``d`` is a directory
+
+``b`` is a block device
+
+ACLs
+----
+
+* Access control lists
+* Not recommended; hard to maintain
+* Typically how other OSes manage permissions
+* Support depends on OS and filesystem
+
+Package Management
+------------------
+
+*Take care of installation and removal of software*
+
+**Core Functionality:**
+
+* Install, Upgrade & uninstall packages easily
+* Resolve package dependencies
+* Install packages from a central repository
+* Search for information on installed packages and files
+* Pre-built binaries (usually)
+* Find out which package provides a required library or file
+
+Popular Linux Package Managers
+------------------------------
+
+* .deb / APT + dpkg (used by Debian, Ubuntu, Linux Mint)
+* .rpm / YUM + rpm (used by RedHat, CentOS, Fedora)
+
+RPM & yum (RedHat, CentOS, Fedora)
+----------------------------------
+
+.. image:: ../_static/rpm.png
+    :align: right
+    :width: 30%
+
+**RPM**
+
+  Binary file format which includes metadata about the package and the
+  application binaries as well.
+
+.. image:: ../_static/yum.png
+    :align: right
+    :width: 30%
+
+**Yum**
+
+  RPM package manager used to query a central repository and resolve RPM
+  package dependencies.
+
+Yum Commands (Redhat, CentOS, Fedora)
+-------------------------------------
+
+.. code-block:: bash
+
+  # Searching for a package
+  $ yum search tree
+
+  # Information about a package
+  $ yum info tree
+
+  # Installing a package
+  $ yum install tree
+
+  # Upgrade all packages to a newer version
+  $ yum upgrade
+
+  # Uninstalling a package
+  $ yum remove tree
+
+  # Cleaning the RPM database
+  $ yum clean all
+
+Apt (Debian, Ubuntu)
+--------------------
+
+.. note:: You can also use aptitude as a front-end to dpkg instead of apt-get.
+
+.. code-block:: bash
+
+  # Update package cache database
+  $ apt-get update
+
+  # Searching for a package
+  $ apt-cache search tree
+
+  # Information about a package
+  $ apt-cache show tree
+
+  # Installing a package
+  $ apt-get install tree
+
+  # Upgrade all packages to a newer version
+  $ apt-get upgrade
+  $ apt-get dist-upgrade
+
+  # Uninstalling a package
+  $ apt-get remove tree
+  $ apt-get purge tree
+
+Dpkg Commands
```

Comment:
Might want to add a section talking about rpm vs yum if we're going to talk about apt vs dpkg.